src/Pact/Types/Capability.hs

```diff
@@ -106,19 +107,25 @@ data CapSlot c = CapSlot
   } deriving (Eq,Show,Ord,Functor,Foldable,Traversable,Generic)
 instance NFData c => NFData (CapSlot c)
 
-data ManagedCapability c = ManagedCapability
-  { _mcInstalled :: CapSlot c
-    -- ^ original installed capability, unmutated, with composed caps.
-  , _mcStatic :: UserCapability
-    -- ^ Cap without mutating
-  , _mcManaged :: PactValue
+data UserManagedCap = UserManagedCap
+  { _umcManagedValue :: PactValue
     -- ^ mutating value
-  , _mcManageParamIndex :: Int
+  , _umcManageParamIndex :: Int
     -- ^ index of managed param value
-  , _mcManageParamName :: Text
+  , _umcManageParamName :: Text
     -- ^ name of managed param value
-  , _mcMgrFun :: Def Ref
+  , _umcMgrFun :: Def Ref
     -- ^ manager function
+  } deriving (Show,Generic)
+instance NFData UserManagedCap
+
+data ManagedCapability c = ManagedCapability
+  { _mcInstalled :: CapSlot c
+    -- ^ original installed capability
+  , _mcStatic :: UserCapability
+    -- ^ Cap without any mutating components (for auto, same as installed)
+  , _mcManaged :: Either Bool UserManagedCap
+    -- ^ either auto-managed or user-managed
```

Comment:
I would prefer if we made this its own sort so that we don't have to hunt in code for what `Left (True|False) | Right _` means, which can get confusing. I propose something like the following: 

```haskell
data Linearity = Consumed | Unconsumed
data ManagedCapSort = OneShot !Linearity | UserManaged !UserManagedCap

data ManagedCapability c ... 
  { ...
  , _mcManaged :: ManagedCapSort
  }
```

Reply:
I went for a literate `Bool` newtype, as it felt like a lot of types otherwise. With the `AutoManagedCap` newtype I think the code reads OK.

Reply:
The problem with a `Bool` newtype in this situation is that it still doesn't give me any information about what `True` or `False` mean. Does `True` mean "Fired"? That's what I'd assume at first glance. Writing your own booleans with expressive pattern alts is cheap and optimizes to `Bool` anyway, so we can be really clear about what the cases of a one-shot cap are.

Reply:
As this already deployed, we can address these changes later.